providers/fips/self_test_keygen_pairwise.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h> /* memcmp */
```

Comment:
Because you're using publicaly deprecated low level functions, you need to add this above all other includes:

``` C
/*
 * RSA, ECDSA and DSA low level APIs are deprecated for public use, but still ok for
 * internal use.
 */
#include "internal/deprecated.h"
```